dlrover/python/util/function_util.py

```diff
@@ -55,6 +56,31 @@ def wrapped(*args, **kwargs):
     return decorator
 
 
+def timeout_concurrent(secs=-1, callback_func=None):
+    if callback_func is None:
+        if secs <= 0:
+            timeout_secs_value = TIMEOUT_MAX
+        else:
+            timeout_secs_value = secs
+    else:
+        timeout_secs_value = callback_func()
+
+    def decorator(func):
+        @functools.wraps(func)
+        def wrapped(*args, **kwargs):
+            with futures.ThreadPoolExecutor(max_workers=1) as executor:
```

Comment:
the parallelism should be defined by the input args by decorator

Reply:
This thread pool is only used to execute the function to monitor the timeout. Therefore, It is unnecessary to have more than 1 thread in the pool.